rpc/rpc.go

```diff
@@ -77,28 +77,32 @@ type Conn struct {
 	// bgcancel cancels bgctx.  Callers MUST hold mu.
 	bgcancel context.CancelFunc
 	// tasks block shutdown.
-	tasks sync.WaitGroup
+	tasks  sync.WaitGroup
+	closed chan struct{} // closed when shutdown() returns
 
 	// The underlying transport. Only the reader & writer goroutines
 	// may do IO, but any thread may call NewMessage().
 	transport Transport
 
 	sender mpsc.Queue[asyncSend]
 
-	// mu protects all the following fields in the Conn.
-	mu      sync.Mutex
-	closing bool          // used to make shutdown() idempotent
-	closed  chan struct{} // closed when shutdown() returns
-
-	// Tables
-	questions  []*question
-	questionID idgen
-	answers    map[answerID]*answer
-	exports    []*expent
-	exportID   idgen
-	imports    map[importID]*impent
-	embargoes  []*embargo
-	embargoID  idgen
+	mu sync.Mutex // mu protects all the following fields in lk.
```

Comment:
Minor suggestion, maybe embed `sync.Mutex` directly into `lk`, and rename the latter to `tables`?

```go
lk struct {
	sync.Mutex

	closing bool // used to make shutdown() idempotent

	// Tables
	questions  []*question
	questionID idgen
	answers    map[answerID]*answer
	exports    []*expent
	exportID   idgen
	imports    map[importID]*impent
	embargoes  []*embargo
	embargoID  idgen
}
```

This would have the benefit of:

1. grouping the mutex with the fields it protects
2. correctly labelling the set of tables (it's not really a "lock", but a "table that needs locking")
3. being a bit clearer at the call-site, I think.

Regarding point 3, we would have calls that look like this:

```go
c.tables.Lock()
defer c.tables.Unlock()

c.tables.questions[0] // ...
```

Thoughts?

Reply:
Not sure I'm a fan of `tables`; there are things in there other than tables. Also I shortened `locked` -> `lk` because it came up often enough that the verbosity was starting to get annoying...

I think embedding the mutex is a good idea though, I'll do that.

Reply:
(done)

Reply:
Hm, it's *just* closing that's extra though -- I wonder if it would be cleaner to:

- Actually define this as its own type, `tables`
- Delete `closing`
- Make the field a pointer
- Set it to nil when shutdown starts

...since we already nil out all these at the top of shutdown anyway.

I don't know that I want to do the functional changes in the same PR though. Thoughts?

Reply:
...If we did this we might want to move the mutex back out of tables though, since there's otherwise no way to access it atomically...

Reply:
I actually think that makes sense, including the moving of the mutex back out of tables.  Also agree with doing this in a separate PR.

Is the current state something you can work with for the next few hours, or have we made a mess of things?  If it's the former, I say we merge it and revisit as soon as needed.

Reply:
This is I think a reasonable place to cut it; merge away.